src/internal_telemetry/allocations/mod.rs

```diff
@@ -1,9 +1,53 @@
 //! Allocation tracking exposed via internal telemetry.
 
 mod allocator;
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Mutex,
+    },
+    thread,
+    time::Duration,
+};
+
+use arr_macro::arr;
+use metrics::gauge;
+use rand_distr::num_traits::ToPrimitive;
 
 use self::allocator::Tracer;
-pub(crate) use self::allocator::{AllocationGroupId, AllocationLayer, GroupedTraceableAllocator};
+
+pub(crate) use self::allocator::{
+    without_allocation_tracing, AllocationGroupId, AllocationLayer, GroupedTraceableAllocator,
+};
+
+use crossbeam_utils::CachePadded;
+
+const NUM_GROUPS: usize = 256;
+/// These arrays represent the allocations and deallocations for each group.
+/// We pad each Atomic to reduce false sharing effects.
+static GROUP_MEM_ALLOCS: [CachePadded<AtomicU64>; NUM_GROUPS] =
+    arr![CachePadded::new(AtomicU64::new(0)); 256];
+static GROUP_MEM_DEALLOCS: [CachePadded<AtomicU64>; NUM_GROUPS] =
+    arr![CachePadded::new(AtomicU64::new(0)); 256];
+
+// By using the Option type, we can do statics w/o the need of other creates such as lazy_static
+struct GroupInfo {
+    component_kind: Option<String>,
+    component_type: Option<String>,
```

Comment:
Looking a little deeper, this value is always derived from `&'static str`, which wouldn't need to be owned. That would allow skipping one allocation and clone when using this.

Reply:
I assumed we needed ownership given that things such as component id and typetag are not ```&'static```